services/actions/notifier_helper.go

```diff
@@ -400,6 +395,11 @@ func handleSchedules(
 		}
 	}
 
+	if len(detectedWorkflows) == 0 {
```

Comment:
Not your fault, but I just noticed a huge memory/performance "leak" in the code above this line while trying to understand why it fixes the problem.
I think it is quite wasteful to query and discard all schedules, when a `Has` check would suffice.

Reply:
Could use the `CountSchedules` method instead.